src/Scripts/Game/Features/Gathering/EL_GatherAction.c

```diff
@@ -5,31 +5,114 @@ class EL_GatherAction : ScriptedUserAction
+	// User has performed the action
+	// play a pickup sound and then add the correct amount to the users inventory
 	override void PerformAction(IEntity pOwnerEntity, IEntity pUserEntity)
 	{
-		// Play sound
-		auto replication = RplComponent.Cast(pOwnerEntity.FindComponent(RplComponent));
+		RplComponent replication = RplComponent.Cast(pOwnerEntity.FindComponent(RplComponent));
 		
-		//Todo: Replace SCR_InventoryStorageManagerComponent RPL setup with own function or better synced sound alternative
-		auto inventoryManager = SCR_InventoryStorageManagerComponent.Cast(pUserEntity.FindComponent(SCR_InventoryStorageManagerComponent));
-		inventoryManager.PlayItemSound(replication.Id(), "SOUND_PICK_UP");
+		m_InventoryManager.PlayItemSound(replication.Id(), "SOUND_PICK_UP");
 		
 		//Spawn item
-		inventoryManager.TrySpawnPrefabToStorage(m_GatherItemPrefab);
+		for (int i = 0; i < m_AmountGathered; i++)
+			m_InventoryManager.TrySpawnPrefabToStorage(m_GatherItemPrefab);
 		
 		//Show hint what to do with the gathered item
 		EL_GameModeRoleplay.GetInstance().ShowInitalTraderHint();
+		
+		if (m_DelayTimeMilliseconds > 0) // If item has a delay between uses
+		{
+			SetCannotPerformReason(string.Format("Please wait %1 seconds", m_DelayTimeMilliseconds / 1000));
+			
+			m_CanBePerformed = false;
+			GetGame().GetCallqueue().CallLater(ToggleCanBePerformed, m_DelayTimeMilliseconds);	
+		}
 	}
 	
+	//------------------------------------------------------------------------------------------------
+	// Formats name for action when hovering
 	override bool GetActionNameScript(out string outName)
 	{
 		outName = string.Format("Gather %1", m_GatherItemDisplayName);
 		return true;
 	}
 	
+	//------------------------------------------------------------------------------------------------
+	// Checks if a required item has been set in the Editor
+	// If so, check if its in the users inventory/hands depending on settings set
 	override bool CanBePerformedScript(IEntity user)
  	{
-		return true;
+		if (!m_CanBePerformed)
+			return false;
+		
+		m_InventoryManager = SCR_InventoryStorageManagerComponent.Cast(user.FindComponent(SCR_InventoryStorageManagerComponent));
+		
+		if (!m_RequiredItemPrefab) // If not required we dont need to check anything
+			return true;
+		
+		if (m_CheckInventory) // Just check the inventory
+		{
+			SetCannotPerformReason(string.Format("Requires %1", m_RequiredItemDisplayName));
+			
+			array<BaseInventoryStorageComponent> outStorages = new array<BaseInventoryStorageComponent>();
+			m_InventoryManager.GetStorages(outStorages);
+			
+			foreach (BaseInventoryStorageComponent storage : outStorages)
+			{
+				array<IEntity> outItems = new array<IEntity>();
+				storage.GetAll(outItems);
+				
+				foreach (IEntity item : outItems)
+				{
+					if (item.GetPrefabData().GetPrefabName() == m_RequiredItemPrefab)
+						return true;
+				}
+			}
```

Comment:
You can use `m_InventoryManager.GetItems()` instead of loading the storages and then the items manually :)